### PR TITLE
Random-Sampling unit tests and git actions

### PR DESCRIPTION
**Pull request by FilipJoksimovic29**
Added RandomSampling unit tests for:
 constructor
       Should set correct initial values
       Should revert if avgBlockTimeInSeconds is 0
       Should set correct Hub reference
    name()
       Should return correct name
    version()
       Should return correct version
    setProofingPeriodDurationInBlocks()
       Should revert if durationInBlocks is 0
       Should revert if called by non-contract
       setW1() and W1 getter
       Should update W1 correctly and revert for non-owners
       setW2() and W2 getter
       Should update W2 correctly and revert for non-owners
    setAvgBlockTimeInSeconds()
       Should update avgBlockTimeInSeconds and revert for non-owners
       Should revert if blockTimeInSeconds is 0
      
   Added RandomSampling storage tests for:
      Scoring System
       Should increment and get epoch node valid proofs count
       Should add to node score and get node score
       Should accumulate delegator scores correctly
    Initialization
       Should have correct name and version
       Should set the initial parameters correctly
       Should set correct Chronos reference and epoch on initialize
       Should only apply latest epoch on multiple initialize calls
       Should set correct initial values
       Should update effectiveEpoch to current epoch after initialize
       Should revert if proofingPeriodDurationInBlocks is 0
       Should set correct CHUNK_BYTE_SIZE constant
    Access Control
       Should revert contact call if not called by Hub
       Should revert contact call on onlyContract modifiers
       Should allow access when called by Hub
       Should allow access when called by registered contract
    Proofing Period Management
       Should return the correct proofing period status
       Should update start block correctly for different period scenarios
       Should return correct historical proofing period start
       Should return correct active proof period
       Should pick correct proofing period duration based on epoch
       Should return correct proofing period duration based on epoch history
       Should return same block when no period has passed
       Should return correct status for different block numbers
    Challenge Handling
       Should set and get challenge correctly
       Should handle multiple challenges and updates correctly
    Proofing Period Duration Management
       Should correctly track pending proofing period duration
       Should handle multiple proofing period durations correctly
       Should replace pending duration correctly
    Delegator Rewards Management
       Should track delegator rewards claimed status correctly
       Should handle multiple delegators rewards claimed status
       Should maintain separate claimed status for different epochs
    Edge Cases
       Should revert if no matching duration in blocks found
       Should handle large number of proofing durations correctly 
       
GitHub Actions are added to run these unit and integration tests whenever a push or pull request is made to the feature/random-sampling branch.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Random Sampling Tests
+
+on:
+  pull_request:
```
Comment (Mihajlo-Pavlovic) [inline .github/workflows/test.yml:4]: Should this maybe run on every PR? Why limit to only these branches

@Bojan131

**Review by Mihajlo-Pavlovic — CHANGES_REQUESTED** [review]
Nice unit test, gj!

**Review by branarakic — APPROVED** [review]
Good job @FilipJoksimovic29
